### PR TITLE
Prepare account components for reuse in the profile

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- open your account
- play around with the account page features, all should work as before

**Comment by huwshimi** [comment]
[Prep account components for sharing with profile](https://trello.com/c/lZNMMxLT/145-prep-account-components-for-sharing-with-profile)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8369/

File: jujugui/static/gui/src/app/init/component-renderers-mixin.js [hunk 1]
```diff
@@ -715,58 +717,43 @@ Browser: ${navigator.userAgent}`
...
-        removePaymentMethod={
-          this.payment && this.payment.removePaymentMethod.bind(this.payment)}
-        revokeCloudCredential={
-          controllerAPI.revokeCloudCredential.bind(controllerAPI)}
+        initUtils={shapeup.addReshape({
+          generateCloudCredentialName: initUtils.generateCloudCredentialName.bind(initUtils),
+          getCloudProviderDetails: initUtils.getCloudProviderDetails.bind(initUtils),
+          validateForm: initUtils.validateForm.bind(initUtils)
+        })}
+        payment={payment && shapeup.addReshape({
```
Comment (frankban) [inline jujugui/static/gui/src/app/init/component-renderers-mixin.js]: Can't you just use shapeup.fromShape here?

Comment (frankban) [reply jujugui/static/gui/src/app/init/component-renderers-mixin.js]: Also above in the other similar changes?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8372/

File: jujugui/static/gui/src/app/init/component-renderers-mixin.js (hunk 1 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/init/component-renderers-mixin.js]: Ah cool, very handy.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8373/

**Comment by huwshimi** [comment]
Thanks @frankban :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**